### PR TITLE
fix: elmo load/save path  #710 and add/fix: add new events for train processes

### PR DESCRIPTION
**Pull request by kudep**
fix elmo load/save path   #710
add new events of nn_trainer class:
- before_validation
- before_train
- before_log

### CODE REVIEW

File: deeppavlov/models/elmo/elmo2tfhub.py [hunk 1]
```diff
@@ -198,6 +207,8 @@ def export2hub(weight_file, hub_dir, options):
 
             with tf.Session() as sess:
                 sess.run(tf.global_variables_initializer())
+                if hub_dir.exists():
+                    rmdir(hub_dir)
```
Comment (yoptar) [inline deeppavlov/models/elmo/elmo2tfhub.py]: https://docs.python.org/3/library/shutil.html#shutil.rmtree

File: deeppavlov/models/elmo/elmo.py [hunk 2]
```diff
@@ -272,6 +273,7 @@ def __init__(self,
         self._build_model(train=False, epoch=load_epoch_num)
 
         self.save()
+        self.load_path = self.save_path
```
Comment (yoptar) [inline deeppavlov/models/elmo/elmo.py]: Please comment on this or use a different variable for reloading model while training

File: deeppavlov/models/elmo/elmo2tfhub.py (hunk 1 above)
Comment (kudep) [reply deeppavlov/models/elmo/elmo2tfhub.py]: it's updated

File: deeppavlov/models/elmo/elmo.py [hunk 3]
```diff
@@ -272,6 +273,9 @@ def __init__(self,
         self._build_model(train=False, epoch=load_epoch_num)
 
         self.save()
+        # after building the model and saving to the specified save path
+        # you need to change the way to load the checkpoint
```
Comment (yoptar) [inline deeppavlov/models/elmo/elmo.py]: `you need to change` sounds like a requirement for a user to do something
```suggestion
        # change the way to load intermediate checkpoints
```

**Pull request merged**